### PR TITLE
Rs/sparse integration test clean 2

### PR DESCRIPTION
**Pull request by robertgshaw2-neuralmagic**
SUMMARY:
Adds support for end-to-end sparsity integration tests for 2:4 and unstructured using the logits comparison method

TEST PLAN:
- Correctness testing by comparing "running dense" vs "running magic_wand" kernels
- Memory usage by comparing the resulting size of the kv cache, and checking that the vllm cache is bigger (and thus the weights are smaller)

### CODE REVIEW

**Review by tlrmchlsmth — APPROVED** [review]
LGTM when green. Could you fill out the PR summary as well before landing?

File: tests/models/test_compressed_memory.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+MODEL_FORMAT_EXTRABLOCKS = [
+    ("nm-testing/OpenHermes-2.5-Mistral-7B-pruned50", "sparse_w16a16", 2000),
+    ("nm-testing/OpenHermes-2.5-Mistral-7B-pruned2.4",
+     "semi_structured_sparse_w16a16", 2000),
+]
+
+
+@pytest.mark.parametrize("model_format_extrablocks", MODEL_FORMAT_EXTRABLOCKS)
+@pytest.mark.parametrize("dtype", ["half"])
```
[the remark this answers is not in the document]
Comment (robertgshaw2-neuralmagic) [reply tests/models/test_compressed_memory.py:23]: going to address in other PR

File: tests/models/test_compressed.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    ("nm-testing/TinyLlama-1.1B-Chat-v1.0-pruned2.4",
+     "semi_structured_sparse_w16a16"),
+    ("nm-testing/OpenHermes-2.5-Mistral-7B-pruned50", "sparse_w16a16"),
+    ("nm-testing/OpenHermes-2.5-Mistral-7B-pruned2.4",
+     "semi_structured_sparse_w16a16"),
+]
+
+
+@pytest.mark.parametrize("model_format_pairs", MODEL_FORMAT_PAIRS)
+@pytest.mark.parametrize("dtype", ["half"])
```
[the remark this answers is not in the document]
Comment (robertgshaw2-neuralmagic) [reply tests/models/test_compressed.py:24]: going to address in other PR

**Review by andy-neuma — APPROVED** [review]
checked `testmo` run and the "test compressed" ran ... thanks!

**Review by mgoin — APPROVED** [review]
